core/templates/dev/head/services/date-time-format.service.spec.ts

```diff
@@ -62,4 +62,13 @@ describe('datetimeformatter', () => {
       df.getLocaleAbbreviatedDatetimeString(
         NOW_MILLIS - 365 * 24 * 60 * 60 * 1000)).toBe('11/21/13');
   });
+
+  it('should provide correct date format MM/DD/YYY string', () => {
```

Comment:
Having a lot of logic to compute the expected value isn't really good practice. So, I wanted to check: have you investigated if there are ways to change the default locale on the browser while it is being used for testing? It would be better to set the environment to what you need it to be (at the start of the test) and then unset it when the test has completed.

This may or may not be possible, but it should at least be investigated for feasibility.

Reply:
There isn't a way to change the default locale on the browser. I've tried to test it evaluating day, month and year but I've searched and there is no way to construct a date based on a locale, and I didn't find a way to get the default locale too. So without this features, the tests were failing since each user can have different dates formats.
The way I'm testing now is the best approach I've found so far. Can you explain more about testing the implementation more than the interface? I think I didn't get it.

Reply:
OK, let's keep it as you have it, then. The only thing I would say is to please add a detailed comment to this part of the code explaining that you're doing this because different browsers have different locales, and you can't find a way to set the locale on the browser in Karma tests. Be as verbose and clear as possible so that other devs have full context (since, that way, if they can find a way to handle browser locales in the future, they can do so).

For interface vs implementation, often just searching for articles on things like this will help. Here is the second result for "testing interface vs implementation": https://mehulkar.com/blog/unit-testing-interface-vs-implementation/ -- you can find more info by reading other search results for that query.

Reply:
Thanks for sharing the article! I've added the comment, PTAL